### PR TITLE
feat: support versioned URLs

### PR DESCRIPTION
**Pull request by BRKalow**
## "Ready for Review" checklist

- [ ] The Vercel preview link has been added to this PR's description
- [ ] The Asana task has been added to this PR's description
- [ ] This PR's link has been added to the "📐 Pull Request" field on the Asana task
- [ ] This Vercel preview link has been added to the "🔍 Preview" field on the Asana task
- [ ] The description template has been filled in below

---

## Relevant links

- [Preview link](https://dev-portal-git-BRANCH_NAME-hashicorp.vercel.app/PATH_TO_VIEW) 🔎
- [Asana task](https://app.asana.com/0/1201903760348480/1202150565840117/f) 🎟️

## What

Support rendering of versioned docs paths

## Why

For versioned docs support!

## How

This is already baked into the underlying content loader, just needed to set `enableVersionedDocs: true` and pass along the `versions` array. We can use the `versions` array to build out the version select after this.

## Testing

- [ ] Navigate to https://dev-portal-git-brkchore-enable-versioned-docs-urls-hashicorp.vercel.app/waypoint/docs/v0.7.x
- [ ] Assert that the content renders
- [ ] Navigate to https://dev-portal-git-brkchore-enable-versioned-docs-urls-hashicorp.vercel.app/vault/docs/v1.9.x
- [ ] Assert that the content renders
- [ ] Log `__NEXT_DATA__.props.pageProps.versions` in the console
- [ ] Assert that the available versions are in the array

### CODE REVIEW

File: src/layouts/sidebar-sidecar/server.ts [hunk 1]
```diff
@@ -51,6 +51,7 @@ export function getStaticGenerationFunctions<
   const loaderOptions: RemoteContentLoader['opts'] = {
     product: productSlugForLoader,
     basePath: basePathForLoader,
+    enabledVersionedDocs: true,
```
Comment (BRKalow) [inline src/layouts/sidebar-sidecar/server.ts:54]: 🪄

Comment (thiskevinwang) [reply src/layouts/sidebar-sidecar/server.ts:54]: 💭 Any reason to **not** use `process.env.ENABLE_VERSIONED_DOCS` or `isVersionedDocsEnabled`?

**Review by thiskevinwang — APPROVED** [review]
Makes sense to me!

File: src/layouts/sidebar-sidecar/server.ts (hunk 1 above)
Comment (BRKalow) [reply src/layouts/sidebar-sidecar/server.ts:54]: Hmm, I don't see any reason _not_ to hardcode it at the moment. The intent behind that env var was mostly just to facilitate the rollout. So I guess it's tech debt now 🙃

**Pull request merged**